x-pack/agent/docs/agent_configuration_example.yml

```diff
@@ -0,0 +1,216 @@
+outputs:
+  default:
+    api_key: VuaCfGcBCdbkQm-e5aOx:ui2lp2axTNmsyakw9tvNnw
+    hosts: ["localhost:9200"]
+    ca_sha256: "7lHLiyp4J8m9kw38SJ7SURJP4bXRZv/BNxyyXkCcE/M="
+    # Not supported at first
+    queue:
+      type: disk
+
+  long_term_storage:
+    api_key: VuaCfGcBCdbkQm-e5aOx:ui2lp2axTNmsyakw9tvNnw
+    hosts: ["localhost:9200"]
+    ca_sha256: "7lHLiyp4J8m9kw38SJ7SURJP4bXRZv/BNxyyXkCcE/M="
+    queue:
+      type: disk
+
+  monitoring:
+    api_key: VuaCfGcBCdbkQm-e5aOx:ui2lp2axTNmsyakw9tvNnw
+    hosts: ["localhost:9200"]
+    ca_sha256: "7lHLiyp4J8m9kw38SJ7SURJP4bXRZv/BNxyyXkCcE/M="
+
+settings.monitoring:
+  use_output: monitoring
+
+datasources:
+  # use the nginx package
+  - id?: nginx-x1
+    enabled?: true # default to true
+    title?: "This is a nice title for human"
+    # Package this config group is coming from. On importing, we know where it belongs
+    # The package tells the UI which application to link to
+    package?:
+      name: epm/nginx
+      version: 1.7.0
+    namespace?: prod
+    constraints?:
+      # Contraints look are not final
+      - os.platform: { in: "windows" }
+      - agent.version: { ">=": "8.0.0" }
+    use_output: long_term_storage
+    inputs:
+      - type: logs
+        processors?:
+        streams:
+          - id?: {id}
+            enabled?: true # default to true
+            dataset: nginx.acccess
+            paths: /var/log/nginx/access.log
```

Comment:
we say paths but there is only one. is this an array or should this be a `path`?

Reply:
Well no its should still be plural since we are using go-ucfg, a single string "/var/log" and a slice of strings are equivalent, if you use the following type to unpack. This is true for a lot of things in beats.


``` golang
type C struct {
 Paths []string `config:"path"`
}
```

Reply:
in ucfg yes but in tests we're producing map[string]interface{} tree so i was wondering what's the intention here.

Reply:
This is correct, but they will be reparsed after so that should not be a problem, or I am missing something?

Reply:
no no you're not, i just wanted to have these trees in a form which reflects our intention. so double checking whether this is supposed to be multiple paths or single value as with metricset